packages/core/src/mesh/MeshRenderer.ts

```diff
@@ -19,6 +20,12 @@ export class MeshRenderer extends Renderer implements ICustomClone {
   private static _tangentMacro = Shader.getMacroByName("O3_HAS_TANGENT");
   private static _vertexColorMacro = Shader.getMacroByName("O3_HAS_VERTEXCOLOR");
 
+  /** whether receive shadow */
+  receiveShadow: boolean = false;
+
+  /** whether cast shadow */
+  castShadow: boolean = false;
+
```

Comment:
- Maybe any Renderer should have shadow in the future, `receiveShadow ` and `castShadow ` maybe in `Renderer`

- 
![image](https://user-images.githubusercontent.com/4967545/183870082-22660a36-ab04-4088-b0ce-8ba807c169d8.png)
Maybe use `receiveShadows` and `castShadows` is better